### DISCUSSION TITLE
How to add "content_security_policy" to manifest?

### DISCUSSION THREAD

**Discussion by KULTI1995** [wxt-dev/wxt#1055] [Q&A]
How to add to manifest?
"content_security_policy": "script-src 'self' https://www.google-analytics.com; object-src 'self'"

Have you ever implemented tracking in browser extensions? Do you have an example of implementation?
Tracking installations for Google Ads and other actions.

**Comment by aklinker1** [comment]
> How to add to manifest?

- https://wxt.dev/get-started/configuration.html#manifest-json
- https://developer.chrome.com/docs/extensions/reference/manifest/content-security-policy

```ts
export default defineConfig({
  manifest: {
    content_security_policy: {
      extension_pages: "script-src 'self' https://www.google-analytics.com/; object-src 'self'"
    },
  },
});
```

> Have you ever implemented tracking in browser extensions? Do you have an example of implementation? Tracking installations for Google Ads and other actions.

I have never gotten conversion tracking setup with Google Ads, no. AFAIK, it's impossible to do. 

If you want to track analytics events, you can use the [Measurement Protocol](https://developers.google.com/analytics/devguides/collection/protocol/ga4) or just wait for https://github.com/wxt-dev/wxt/pull/790, which will support the measurement protocol.
